pkg/controller/jobframework/reconciler.go

```diff
@@ -285,7 +284,11 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	if wl == nil {
 		err := r.handleJobWithNoWorkload(ctx, job, object)
 		if err != nil {
-			log.Error(err, "Handling job with no workload")
+			if IsUnretryableError(err) {
```

Comment:
in which cases `handleJobWithNoWorkload` returns an unretriable error?

Reply:
Each time `ConstructComposableWorkload` which is called [here](https://github.com/epam/kubernetes-kueue/blob/pod_groups_concurrent_reconciles/pkg/controller/jobframework/reconciler.go#L679) returns an unretriable error.

https://github.com/epam/kubernetes-kueue/blob/pod_groups_concurrent_reconciles/pkg/controller/jobs/pod/pod_controller.go#L604-L608

https://github.com/epam/kubernetes-kueue/blob/pod_groups_concurrent_reconciles/pkg/controller/jobs/pod/pod_controller.go#L616-L620

Maybe it's not really necessary to have this if. The idea here is that an unretriable error should be logged without a traceback. This could also be done by using [this function](https://pkg.go.dev/github.com/go-logr/logr#Logger.WithCallStackHelper) as far as I understand. I could look into that so that any log.Error call won't print a traceback for UnrtriableErrors.